### PR TITLE
Updates for CMAC

### PR DESCRIPTION
**Pull request by andrzej-kaczmarek**

### CODE REVIEW

File: hw/mcu/dialog/cmac/src/cmac_timer.c [hunk 1]
```diff
@@ -367,11 +367,17 @@ cmac_timer_slp_disable(uint32_t exp_ticks)
 bool
 cmac_timer_slp_update(void)
 {
-    if (g_cmac_shared_data.lp_clock_freq == g_cmac_timer_slp.freq) {
+    uint32_t lp_clock_freq;
+
+    cmac_shared_lock();
```
Comment (kasjer) [inline hw/mcu/dialog/cmac/src/cmac_timer.c:372]: locking read of single value seems like overkill

File: hw/mcu/dialog/cmac/src/cmac_isr.c [hunk 2]
```diff
@@ -20,14 +20,27 @@
...
     if (CMAC->CM_EXC_STAT_REG & CMAC_CM_EXC_STAT_REG_EXC_SYS2CMAC_Msk) {
+        cmac_shared_lock();
+        pending_ops = g_cmac_shared_data.pending_ops;
+        g_cmac_shared_data.pending_ops = 0;
+        cmac_shared_unlock();
+
         cmac_mbox_read();
         cmac_rand_read();
+
+        if (pending_ops & CMAC_PENDING_OP_LP_CLK) {
```
Comment (kasjer) [inline hw/mcu/dialog/cmac/src/cmac_isr.c:40]: Where CMAC_PENDING_OP_LP_CLK is defined?

Comment (andrzej-kaczmarek) [reply hw/mcu/dialog/cmac/src/cmac_isr.c:40]: https://github.com/apache/mynewt-nimble/pull/874

**Review by kasjer — APPROVED** [review]
LGTM

**Pull request merged**